src/app/main/component/shared/components/form-base/form-base.component.ts

```diff
@@ -84,7 +84,7 @@ export class FormBaseComponent implements ComponentCanDeactivate {
             } else {
               this.router.navigate([this.previousPath]);
             }
-          } else if (isUbsOrderSubmit) {
+          } else if (isUbsOrderSubmit && confirm === null) {
             this.cancelUBSwithoutSaving();
           }
```

Comment:
I woud refactor subscribe logic:

const currentUrl = this.router.url;
const isUBS = currentUrl.includes('ubs/order');

if (confirm) {
  this.areChangesSaved = true;
}

if (confirm && isUbsOrderSubmit) {
  this.router.navigate(['ubs', 'confirm']);
}

if (confirm && isUBS) {
  this.cancelUBSwithoutSaving();
}

if (confirm && !isUbsOrderSubmit && !isUBS) {
  this.router.navigate([this.previousPath]);
}

if (!confirm && isUbsOrderSubmit) {
  this.cancelUBSwithoutSaving();
}

Reply:
Thanks. I partially changed the code, like on the refactor.